modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1141,6 +1147,42 @@ private void _registerTransactionCallbackForCluster(
 		}
 	}
 
+	private void _setTitleObjectFieldId(
+			ObjectDefinition objectDefinition, String titleObjectFieldName)
+		throws PortalException {
+
+		List<ObjectField> objectFields =
+			_objectFieldPersistence.findByObjectDefinitionId(
+				objectDefinition.getObjectDefinitionId());
+
+		if (StringUtil.equals(titleObjectFieldName, StringPool.BLANK)) {
+			for (ObjectField objectField : objectFields) {
+				if (!StringUtil.equals(objectField.getName(), "id")) {
+					continue;
+				}
+
+				objectDefinition.setTitleObjectFieldId(
+					objectField.getObjectFieldId());
+
+				return;
+			}
+		}
+
+		for (ObjectField objectField : objectFields) {
+			if (StringUtil.equals(
+					titleObjectFieldName, objectField.getName())) {
+
+				_validateObjectFieldId(
+					objectDefinition, objectField.getObjectFieldId());
+
+				objectDefinition.setTitleObjectFieldId(
+					objectField.getObjectFieldId());
+
+				return;
+			}
+		}
```

Comment:
@p-albuquerque why don't you use the `_objectFieldPersistence.findByODI_N` (find by `objectDefinitionId` and field `name`) instead? I would do something like:

```
if (StringUtil.equals(titleObjectFieldName, StringPool.BLANK)) {
	titleObjectFieldName = "id";
}
  
ObjectField objectField = _objectFieldPersistence.findByODI_N(
	objectDefinition.getObjectDefinitionId(), titleObjectFieldName);
				
if (objectDefinition.isSystem()) {
	_validateObjectFieldId(objectDefinition, objectField.getObjectFieldId());
}
      
objectDefinition.setTitleObjectFieldId(objectField.getObjectFieldId());
```

Reply:
Sounds good! I will change this. Thanks